docs/remote/devcontainerjson-reference.md

```diff
@@ -104,6 +106,20 @@ Variables can be referenced in certain string values in `devcontainer.json` in t
 | `${localWorkspaceFolderBasename}` | Any | Name of the local folder that was opened in VS Code (that contains `.devcontainer/devcontainer.json`).<br /><br />⚠️ Not yet supported when using Clone Repository in Container Volume. |
 | `${containerWorkspaceFolderBasename}` | Any | Name of the folder where the workspace files can be found in the container. |
 
+## Lifecycle scripts
+
+When creating and working with a dev container, you may need different commands to be run at different points in the container's lifecycle. The below are properties that shape how the container is created and takes shape over time, and the order in which they are run (i.e. `onCreateCommand` will run after `initializeCommand`).
+
+| Property | Description |
+|----------|-------------|
+| `initializeCommand` | A command to run locally before anything else. If this is a single string, it will be run in a shell. If this is an array of strings, it will be run as a single command without shell. |
+| `onCreateCommand` | A command to run when creating the container. If this is a single string, it will be run in a shell. If this is an array of strings, it will be run as a single command without shell. <br> You may want to have some script steps happen before VS Code attaches (`onCreateCommand`) and some after (`postCreateCommand`). But, you can change this with `waitFor` to include even waiting for `postStartCommand` if you have something you want to happen every time. |
+| `updateContentCommand` | Reserved for future use. |
```

Comment:
`onCreateCommand` is similar in that it does not add anything to `postCreateCommand` until we get the prebuild feature. Maybe document both as 'reserved'? Or explain a bit the future use? Have we added them to the JSON schema too early? @Chuxel

Reply:
By default we should `waitFor` the `onCreateCommand` - so there is value here in that you can have something run that VS Code should wait for, and then something else that can happen after.  `updateContentCommand` is the weird one.

Reply:
@Chuxel `waitFor` currently defaults to `updateContentCommand`. Let me know if that should change.

Reply:
I tried to word the doc in a way that could apply to any default (`By default, you may have VS Code `waitFor` the `onCreateCommand`...`), but if we'd rather change the wording here and/or default of `updateContentCommand`, happy to adjust.

Reply:
Yeah updateContentCommand is fine as the default. So we could probably say:

By default, the `waitFor`  property is set to `updateContentCommand`. This allows you to use  use `onCreateCommand` or `updateContentCommend` for steps to it need to have happen before VS Code connects while still using `postCreateCommand` for steps that can happen behind the scenes afterwards.

We could probably say that it `updateContentCommand` is reserved for future scenarios where the source code may be updated sometime during the create process to disambiguate its purpose a bit without signaling specifically its purpose.

Reply:
I like your wording ideas, and the way to disambiguate `updateContentCommand` for now. Updated both the `waitFor` and `updateContentCommand` descriptions, let me know what you think.

Reply:
Looks great - one other thought is we could include an anchor link to the lifecycle section in the devcontainer.json propeprty section to help raise visibility to that since I do think its quite useful.